### PR TITLE
Various Improvements

### PR DESCRIPTION
**Pull request by leo-ship-it**
- Return the amount converted.
 - Improve the dashboard: show urls in Spawned Neurons section and improve how the Status is shown in Withdrawal Requests table.
 - Switch from \`get_pending_proposals\` to \`list_proposals\`.
 - \`get_airdrop_allocation\` now takes an optional Principal as a parameter.

### CODE REVIEW

File: water_neuron/src/conversion.rs [hunk 1]
```diff
@@ -79,6 +77,7 @@ pub async fn nicp_to_icp(arg: ConversionArg) -> Result<WithdrawalSuccess, Conver
                 Ok(WithdrawalSuccess {
                     withdrawal_id,
                     block_index,
+                    icp_amount: Some(icp_due),
```
Comment (EnzoPlayer0ne) [inline water_neuron/src/conversion.rs:80]: why are we now tracking the `icp_amount`? Are we doing something with it in the front-end?

File: water_neuron/src/dashboard.rs [hunk 2]
```diff
@@ -300,10 +300,15 @@ fn construct_to_disburse_table() -> String {
...
+                let neuron_string = format!(
+                    "<a href=\"{}\" target=\"_blank\">{}</a>",
+                    neuron_id.to_dashboard_link(),
+                    neuron_id.id
+                );
                 write!(
                     buf,
-                    "<tr><td>{:?}</td><td>{}</td></tr>",
-                    neuron_id, disburse_request.receiver,
+                    "<tr><td>{neuron_string}</td><td>{}</td></tr>",
```
Comment (EnzoPlayer0ne) [inline water_neuron/src/dashboard.rs:310]: let's move it out of `{}`, to stay more consistent?

Comment (EnzoPlayer0ne) [reply water_neuron/src/dashboard.rs:310]: Why did we remove the debug sign as well?

File: water_neuron/src/proposal.rs [hunk 3]
```diff
@@ -1,33 +1,49 @@
...
+const BATCH_SIZE_LIMIT: u32 = 50;
+const REWARD_STATUS_ACCEPT_VOTES: i32 = 1;
+const REWARD_STATUS_READY_TO_SETTLE: i32 = 2;
+
 pub async fn mirror_proposals() -> Result<(), String> {
     let subaccount = compute_neuron_staking_subaccount_bytes(self_canister_id(), 0).to_vec();
 
-    match get_pending_proposals().await {
+    let list_proposals_args = ListProposalInfo {
+        limit: BATCH_SIZE_LIMIT,
```
Comment (EnzoPlayer0ne) [inline water_neuron/src/proposal.rs:22]: what if the number of proposals we selected exceeds the `BATCH_SIZE_LIMIT`?

Comment (EnzoPlayer0ne) [reply water_neuron/src/proposal.rs:22]: The default argument is 100
```
pub struct ListProposalInfo {
    /// Limit on the number of \[ProposalInfo\] to return. If no value is
    /// specified, or if a value greater than 100 is specified, 100
    /// will be used.
    #[prost(uint32, tag = "1")]
    pub limit: u32,
```
I think we should keep it that way.

File: water_neuron/src/proposal.rs [hunk 4]
```diff
@@ -1,33 +1,49 @@
...
+        before_proposal: None,
+        exclude_topic: vec![],
+        include_reward_status: vec![REWARD_STATUS_ACCEPT_VOTES, REWARD_STATUS_READY_TO_SETTLE],
+        ..Default::default()
```
Comment (EnzoPlayer0ne) [inline water_neuron/src/proposal.rs:26]: Should we add 
```
    /// Omits "large fields" from the response. Currently only omits the
    /// `logo` and `token_logo` field of CreateServiceNervousSystem proposals. This
    /// is useful to improve download times and to ensure that the response to the
    /// request doesn't exceed the message size limit.
    #[prost(bool, optional, tag = "7")]
    pub omit_large_fields: ::core::option::Option<bool>,
``` 
as well?

File: water_neuron/src/conversion.rs (hunk 1 above)
Comment (leo-ship-it) [reply water_neuron/src/conversion.rs:80]: It's to improve interactions with canisters.

**Review by uk4z — APPROVED** [review]
Gfm

File: water_neuron/src/dashboard.rs (hunk 2 above)
Comment (leo-ship-it) [reply water_neuron/src/dashboard.rs:310]: We are now displaying a link to the dashboard.

File: water_neuron/src/proposal.rs [hunk 5]
```diff
@@ -1,33 +1,49 @@
...
+    TaskType, DEBUG, INFO, ONE_HOUR_SECONDS, RETRY_DELAY_VOTING, SEC_NANOS,
 };
 use ic_canister_log::log;
 use ic_sns_governance::pb::v1::manage_neuron::Command as CommandSns;
 use ic_sns_governance::pb::v1::manage_neuron_response::Command as CommandSnsResponse;
 
+const BATCH_SIZE_LIMIT: u32 = 50;
+const REWARD_STATUS_ACCEPT_VOTES: i32 = 1;
+const REWARD_STATUS_READY_TO_SETTLE: i32 = 2;
+
```
Comment (EnzoPlayer0ne) [inline water_neuron/src/proposal.rs:17]: Should we also add 
```
  PROPOSAL_REWARD_STATUS_UNSPECIFIED = 0;
```

Comment (leo-ship-it) [reply water_neuron/src/proposal.rs:17]: This one is never used.

File: water_neuron/src/proposal.rs (hunk 4 above)
Comment (leo-ship-it) [reply water_neuron/src/proposal.rs:26]: Let's omit them.

**Review by EnzoPlayer0ne — APPROVED** [review]

**Pull request merged**